### DISCUSSION TITLE
Issue using Twitter API SDK after authenticating with Twitter Provider

### DISCUSSION THREAD

**Discussion by jacklynch00** [nextauthjs/next-auth#5969] [Help]
### Question 💬

I am using the twitter api sdk in a Next.js server-less function and I am currently creating a new twitter `Client` on each request, which works for the first request but fails every time after until I get a new access token.

### Expected behavior

I expect to be able to make multiple requests with the Twitter API without having to re-authenticate and get a new API Access Token.

### How to reproduce ☕️

Step 1 - Authenticate
- Using Next.JS's `next-auth` package, I authenticate with the Twitter API and store the access and refresh tokens in a JWT.

Step 2 - Use Next.js server-less function to create Twitter Client and make a request
- I then make a request to the next.js server-less function which get the users JWT, grabs the needed information (tokens, scope, etc.), and makes the request to the Twitter API

```
export default async function twitterTweet(req: NextApiRequest, res: NextApiResponse) {
	const session = await getSession({ req });
	const token = await getToken({ req, secret: process.env.NEXTAUTH_SECRET });

	if (!session) {
		return res.status(401).end('Not authenticated');
	}

	const tokenInfo = hasTwitterConfig(token) ? token : null;
	if (!tokenInfo) {
		return res.status(400).end('Access token not provided');
	}

	const twitterClient = initTwitterClient(tokenInfo.twitter);

	const tweet = await twitterClient.tweets
		.createTweet({
			text: 'This is a test tweet',
		});

	return res.status(200).json(tweet);
}
```

Step 3 - Encounter error after making one request
- I am able to successfully complete the above request one time with no issues and I am able to create a tweet. However, if I try and do it again, there is an error thrown by the Twitter SDK indicating an issue with the token (even though it just worked). I am then able to logout and log back in (thus getting a new access token) and successfully complete one request again

```
error - TwitterResponseError
    at request (/Users/jack/code/threadifier/node_modules/twitter-api-sdk/dist/request.js:67:15)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async rest (/Users/jack/code/threadifier/node_modules/twitter-api-sdk/dist/request.js:100:22)
    at async OAuth2User.refreshAccessToken (/Users/jack/code/threadifier/node_modules/twitter-api-sdk/dist/OAuth2User.js:67:22)
    at async OAuth2User.getAuthHeader (/Users/jack/code/threadifier/node_modules/twitter-api-sdk/dist/OAuth2User.js:208:13)
    at async request (/Users/jack/code/threadifier/node_modules/twitter-api-sdk/dist/request.js:48:11)
    at async rest (/Users/jack/code/threadifier/node_modules/twitter-api-sdk/dist/request.js:100:22)
    at async twitterTweet (webpack-internal:///(api)/./pages/api/twitter/tweet/all.ts:32:19)
    at async Object.apiResolver (/Users/jack/code/threadifier/node_modules/next/dist/server/api-utils/node.js:363:9) {
  status: 400,
  statusText: 'Bad Request',
  headers: {
    'cache-control': 'no-cache, no-store, max-age=0',
    connection: 'close',
    'content-disposition': 'attachment; filename=json.json',
    'content-encoding': 'gzip',
    'content-length': '103',
    'content-type': 'application/json;charset=UTF-8',
    date: 'Sun, 04 Dec 2022 20:52:24 GMT',
    perf: '7626143928',
    server: 'tsa_b',
    'set-cookie': 'guest_id_marketing=v1%3A167018714457757711; Max-Age=63072000; Expires=Tue, 03 Dec 2024 20:52:24 GMT; Path=/; Domain=.twitter.com; Secure; SameSite=None, guest_id_ads=v1%3A167018714457757711; Max-Age=63072000; Expires=Tue, 03 Dec 2024 20:52:24 GMT; Path=/; Domain=.twitter.com; Secure; SameSite=None, personalization_id="v1_3Yqtlpgf4Rlqo2sBprv9Cw=="; Max-Age=63072000; Expires=Tue, 03 Dec 2024 20:52:24 GMT; Path=/; Domain=.twitter.com; Secure; SameSite=None, guest_id=v1%3A167018714457757711; Max-Age=63072000; Expires=Tue, 03 Dec 2024 20:52:24 GMT; Path=/; Domain=.twitter.com; Secure; SameSite=None',
    'strict-transport-security': 'max-age=631138519',
    'x-connection-hash': 'ae9d1813834f482cc556e67b27eb51207eb17eeb2f9888e0c8f3cbbb8a19ce80',
    'x-content-type-options': 'nosniff',
    'x-frame-options': 'SAMEORIGIN',
    'x-response-time': '17',
    'x-transaction-id': '5d99f8a156f432bc',
    'x-xss-protection': '0'
  },
  error: {
    error: 'invalid_request',
    error_description: 'Value passed for the token was invalid.'
  },
  page: '/api/twitter/tweet/all'
}
```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by jacklynch00** [comment]
I see this was converted to a discussion from an issue, could someone provide insight into why that was done maybe? Would love to provide more context if needed to better understand the issue I am having above

**Reply by jacklynch00** [reply]
Hello, I would really appreciate any insight here into this problem. I have been stuck on this for some time now and I imagine others would as well.

**Reply by jmcoder1** [reply]
Did you resolve this? I have the same error.

**Comment by anibalsanchez** [comment]
Same issue here. It seems that the problem randomly affects some accounts.

**Comment by jmcoder1** [comment]
I abandoned this library and make the requests using Axios and it now works

**Reply by arslananjum401** [reply]
can you please share how to get first refresh token

**Reply by jmcoder1** [reply]
https://stackoverflow.com/a/76443103/7138457
